src/fidesops/models/privacy_request.py

```diff
@@ -118,15 +161,55 @@ def get_results(self) -> Dict[str, Any]:
         result_prefix = f"{self.id}__*"
         return cache.get_encoded_objects_by_prefix(result_prefix)
 
-    # passive_deletes="all" prevents execution logs from having their privacy_request_id set to null when
-    # a privacy_request is deleted.  We want to retain for record-keeping.
-    execution_logs = relationship(
-        "ExecutionLog",
-        backref="privacy_request",
-        lazy="dynamic",
-        passive_deletes="all",
-        primaryjoin="foreign(ExecutionLog.privacy_request_id)==PrivacyRequest.id",
-    )
+    def trigger_policy_webhook(self, webhook: WebhookTypes) -> None:
+        """Trigger a request to a single customer-defined policy webhook. Raises an exception if webhook response
+        should cause privacy request execution to stop.
+
+        Pre-Execution webhooks send headers to the webhook in case the service needs to send back instructions
+        to halt.  To resume, they use send a request to the reply-to URL with the reply-to-token.
+        """
+        https_connector: HTTPSConnector = get_connector(webhook.connection_config)
+        request_body = SecondPartyRequestFormat(
+            privacy_request_id=self.id,
+            direction=webhook.direction.value,
+            callback_type=webhook.prefix,
+            identities=self.get_cached_identity_data(),
+        )
+
+        headers = {}
+        is_pre_webhook = webhook.__class__ == PolicyPreWebhook
+        response_expected = webhook.direction == WebhookDirection.two_way
+        if is_pre_webhook and response_expected:
+            headers = {
+                "reply-to": f"/privacy-request/{self.id}/resume",
+                "reply-to-token": generate_request_callback_jwe(webhook),
+            }
+
+        logger.info(f"Calling webhook {webhook.key} for privacy_request {self.id}")
+        response: Optional[SecondPartyResponseFormat] = https_connector.execute(
+            request_body.dict(),
+            response_expected=response_expected,
+            additional_headers=headers,
+        )
+        if not response:
+            return
+
+        response_body = SecondPartyResponseFormat(**response)
+
+        # Cache any new identities
+        if response_body.derived_identities and any(
+            [response_body.derived_identities.dict().values()]
+        ):
+            logger.info(
+                f"Updating known identities on privacy request {self.id} from webhook {webhook.key}."
+            )
+            self.cache_identity(response_body.derived_identities)
+
+        # Pause execution if instructed
+        if response_body.halt and is_pre_webhook:
+            raise PrivacyRequestPaused("Halt instruction received.")
```

Comment:
Can this log message contain the privacy request id?

Reply:
Well this isn't a log, it's just the exception i'm raising, and then i catch that exception in PrivacyRequestRunner.run_webhooks, and log there:, but I can add the id to the exception message.

```python
    for webhook in webhooks.order_by(webhook_cls.order):
            try:
                self.privacy_request.trigger_policy_webhook(webhook)
            except PrivacyRequestPaused:
                logging.info(
                    f"Pausing execution of privacy request {self.privacy_request.id}. Halt instruction received from webhook {webhook.key}."
                )
...
```